tensorboard/plugins/debugger_v2/debugger_v2_plugin_test.py

```diff
@@ -278,6 +293,97 @@ def testServeExecutionDigests400ResponseIfRunParamIsNotSpecified(self):
             {"error": "run parameter is not provided"},
         )
 
+    def testServeSourceFileListIncludesThisTestFile(self):
+        _generate_tfdbg_v2_data(self.logdir)
+        run = self._getExactlyOneRun()
+        response = self.server.get(
+            _ROUTE_PREFIX + "/source_files/list?run=%s" % run
+        )
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(
+            "application/json", response.headers.get("content-type")
+        )
+        source_file_list = json.loads(response.get_data())
+        self.assertIsInstance(source_file_list, list)
+        self.assertIn([_HOST_NAME, _CURRENT_FILE_FULL_PATH], source_file_list)
+
+    def testServeSourceFileListWithoutRunParamErrors(self):
+        # Make request without run param.
+        response = self.server.get(_ROUTE_PREFIX + "/source_files/list")
+        self.assertEqual(400, response.status_code)
+        self.assertEqual(
+            "application/json", response.headers.get("content-type")
+        )
+        self.assertEqual(
+            json.loads(response.get_data()),
+            {"error": "run parameter is not provided"},
+        )
+
+    def testServeSourceFileContentOfThisTestFile(self):
+        _generate_tfdbg_v2_data(self.logdir)
+        run = self._getExactlyOneRun()
+        # First, access the source file list, so we can get hold of the index
+        # for this file. The index is required for the request to the
+        # "/source_files/file" route below.
+        response = self.server.get(
+            _ROUTE_PREFIX + "/source_files/list?run=%s" % run
+        )
+        source_file_list = json.loads(response.get_data())
+        index = source_file_list.index([_HOST_NAME, _CURRENT_FILE_FULL_PATH])
+
+        response = self.server.get(
+            _ROUTE_PREFIX + "/source_files/file?run=%s&index=%d" % (run, index)
+        )
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(
+            "application/json", response.headers.get("content-type")
+        )
+        data = json.loads(response.get_data())
+        self.assertEqual(data["host_name"], _HOST_NAME)
+        self.assertEqual(data["file_path"], _CURRENT_FILE_FULL_PATH)
+        with open(__file__, "rt") as f:
```

Comment:
nit: Text-mode is the default, and I’ve never seen it specified
explicitly before (I had to look it up to see what it did). If this
doesn’t do anything, maybe we should drop it, for readability?